File: hw_3_contacts/src/main/java/com/gmail/elnora/fet/hw_3_contacts/EditContactActivity.java
package com.gmail.elnora.fet.hw_3_contacts;

import android.content.Intent;
import android.os.Bundle;
import android.text.InputType;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ImageButton;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

public class EditContactActivity extends AppCompatActivity {
    private ImageButton buttonBack;
    private EditText editTextName;
    private EditText editTextData;
    private Contact contact;
    private Button buttonRemoveContact;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_edit_contact);

        editTextName = findViewById(R.id.editTextName);
        editTextData =  findViewById(R.id.editTextData);
        buttonBack = findViewById(R.id.buttonBack);
        buttonRemoveContact = findViewById(R.id.buttonRemove);

        getContactIntent();
        setContactData();
        setButtonBackListener();
        setButtonRemoveContactListener();
    }

    private void getContactIntent() {
        Intent intent = getIntent();
        contact = (Contact) intent.getSerializableExtra("EDIT_CONTACT");
    }

    private void setContactData() {
        if (contact.getDataType() == DataType.PHONE) {
            editTextData.setHint(R.string.contactPhoneNumber);
            editTextData.setInputType(InputType.TYPE_CLASS_PHONE);
        } else {
            editTextData.setHint(R.string.contactEmail);
            editTextData.setInputType(InputType.TYPE_TEXT_VARIATION_EMAIL_ADDRESS);
        }
        editTextName.setText(contact.getName());
        editTextData.setText(contact.getData());
    }

    private void setButtonBackListener() {
        buttonBack.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                String name = editTextName.getText().toString();
                String data = editTextData.getText().toString();
                Contact changedContact;
                if (contact.getDataType() == DataType.PHONE) {
                    changedContact = new Contact(contact.getId(), name, data, DataType.PHONE);
                } else {
                    changedContact = new Contact(contact.getId(), name, data, DataType.EMAIL);
                }
                Intent intent = new Intent(EditContactActivity.this, MainActivity.class);
                intent.putExtra("CHANGE_CONTACT", changedContact);
                setResult(RESULT_OK, intent);
                finish();
            }
        });
    }

    private void setButtonRemoveContactListener() {
        buttonRemoveContact.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(EditContactActivity.this, MainActivity.class);
                intent.putExtra("REMOVE_CONTACT", contact);
                setResult(RESULT_OK, intent);
                finish();
            }
        });
    }

}

File: hw_5_contacts_database_kotlin/src/main/java/com/gmail/elnora/fet/hw_5_contacts_database_kotlin/EditContactActivity.kt
package com.gmail.elnora.fet.hw_5_contacts_database_kotlin

import android.content.Intent
import android.os.Bundle
import android.text.InputType
import androidx.appcompat.app.AppCompatActivity
import com.gmail.elnora.fet.hw_5_contacts_database_kotlin.data.DataType
import com.gmail.elnora.fet.hw_5_contacts_database_kotlin.database.Contact
import kotlinx.android.synthetic.main.activity_edit_contact.editTextData
import kotlinx.android.synthetic.main.activity_edit_contact.editTextName
import kotlinx.android.synthetic.main.activity_edit_contact.buttonBack
import kotlinx.android.synthetic.main.activity_edit_contact.buttonRemove

class EditContactActivity : AppCompatActivity() {
    private lateinit var contact: Contact

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_edit_contact)
        getContactIntent()
        setContactData()
        setButtonBackListener()
        setButtonRemoveContactListener()
    }

    private fun getContactIntent() {
        val intent = intent
        contact = intent.getSerializableExtra("EDIT_CONTACT") as Contact
    }

    private fun setContactData() {
        if (contact.dataType === DataType.PHONE) {
            editTextData.setHint(R.string.contactPhoneNumber)
            editTextData.inputType = InputType.TYPE_CLASS_PHONE
        } else {
            editTextData.setHint(R.string.contactEmail)
            editTextData.inputType = InputType.TYPE_TEXT_VARIATION_EMAIL_ADDRESS
        }
        editTextName.setText(contact.name)
        editTextData.setText(contact.data)
    }

    private fun setButtonBackListener() {
        buttonBack.setOnClickListener {
            val name = editTextName.text.toString()
            val data = editTextData.text.toString()
            val changedContact: Contact
            changedContact = if (contact.dataType === DataType.PHONE) {
                Contact(contact.id, name, data, DataType.PHONE)
            } else {
                Contact(contact.id, name, data, DataType.EMAIL)
            }
            val intent = Intent(this@EditContactActivity, MainActivity::class.java)
            intent.putExtra("CHANGE_CONTACT", changedContact)
            setResult(RESULT_OK, intent)
            finish()
        }
    }

    private fun setButtonRemoveContactListener() {
        buttonRemove.setOnClickListener {
            val intent = Intent(this@EditContactActivity, MainActivity::class.java)
            intent.putExtra("REMOVE_CONTACT", contact)
            setResult(RESULT_OK, intent)
            finish()
        }
    }
}